src/data-workspace/section-form/section.js

```diff
@@ -46,18 +47,48 @@ export const SectionFormSection = ({
         return Math.max(...groupedTotalColumns)
     }, [data, groupedDataElements])
 
-    const greyedFields = new Set(
-        section.greyedFields.map((greyedField) =>
-            getFieldId(
-                greyedField.dataElement.id,
-                greyedField.categoryOptionCombo.id
-            )
-        )
+    const greyedFields = useMemo(
+        () =>
+            new Set(
+                section.greyedFields.map((greyedField) =>
+                    getFieldId(
+                        greyedField.dataElement.id,
+                        greyedField.categoryOptionCombo.id
+                    )
+                )
+            ),
+        [section.greyedFields]
     )
 
     const filterInputId = `filter-input-${section.id}`
     const headerCellStyles = classNames(styles.headerCell, 'hide-for-print')
 
+    const cells = useMemo(
+        () =>
+            groupedDataElements.map(({ categoryCombo, dataElements }, i) => (
+                <CategoryComboTableBody
+                    key={i} //if disableDataElementAutoGroup then duplicate catCombo-ids, so have to use index
+                    categoryCombo={categoryCombo}
+                    dataElements={dataElements}
+                    filterText={filterText}
+                    globalFilterText={globalFilterText}
+                    maxColumnsInSection={maxColumnsInSection}
+                    renderRowTotals={section.showRowTotals}
+                    renderColumnTotals={section.showColumnTotals}
+                    greyedFields={greyedFields}
+                />
+            )),
+        [
+            section.showRowTotals,
+            section.showColumnTotals,
+            groupedDataElements,
+            filterText,
+            globalFilterText,
+            maxColumnsInSection,
+            greyedFields,
+        ]
+    )
+
```

Comment:
What kind of input did you do to observe the `children` change? 
I'm pretty sure this is not necessary - since react will use the `key`-prop and see that no dom-changes are necessary. The `children`-prop may change in the profiler, but I don't think this should matter. 
I would rather use `React.memo` on the `CategoryComboTableBody` if this kind of optimization is necessary.